### ISSUE TITLE
HTTP-to-HTTPS redirect

### ISSUE THREAD

**Issue by TheNotary** [TheNotary/terraform-google-static-assets#1]
It looks like support for http-to-https redirects has been added to GCP.  Right now you can enable https, but you can't enforce that all requests to http are automatically redirected to https without a little module update.  

History: https://issuetracker.google.com/issues/35904733?pli=1

GUI Demonstration/ tutorial: https://cloud.google.com/load-balancing/docs/https/setting-up-http-https-redirect

A guy using it in terraform: 
- https://github.com/terraform-google-modules/terraform-google-lb-http/blob/v7.0.0/modules/dynamic_backends/main.tf#L100
- https://github.com/terraform-google-modules/terraform-google-lb-http/blob/v7.0.0/modules/dynamic_backends/main.tf#L164

Basically, instead of having one `google_compute_url_map` and two `google_compute_target_http_proxy` and `google_compute_target_https_proxy` resources pointing to it, you instead have two `google_compute_url_map`s where the one for http is just doing the redirect request, and the other is doing the ordinary thing.
